--- original/usr/share/voicecall-ui-jolla/qml/pages/calllog/CallLogItem.qml	2014-04-11 14:47:07.000000000 +0100
+++ patched/usr/share/voicecall-ui-jolla/qml/pages/calllog/CallLogItem.qml	2014-04-11 11:43:22.000000000 +0100
@@ -38,27 +38,55 @@ ListItem {
         }
 
         spacing: Theme.paddingSmall
+        Item {
+            id: nameItem
+            width: parent.width
 
-        Label {
-            id: firstNameText
-            opacity: privateNumber ? 0.5 : 1.0
-            text: people.populated || remoteUid !== undefined
+            Label {
+                id: firstNameText
+                anchors {
+                    left: parent.left;
+                    verticalCenter: parent.verticalCenter
+                }
+                opacity: privateNumber ? 0.5 : 1.0
+                text: people.populated || remoteUid !== undefined
                       ? (!person || (person.primaryName.length === 0 && person.secondaryName.length === 0)
                              ? main.formatNumber(remoteUid)
-                             : person.primaryName)
+                             : person.primaryName+" ")
                       : ""
+                color: highlighted ? Theme.highlightColor : primaryColor
+                truncationMode: TruncationMode.Fade
+                width: Math.min(implicitWidth, parent.width)
+            }
+            Label {
+                id: secondaryNameText
+                anchors {
+                    left: firstNameText.right;
+                    verticalCenter: parent.verticalCenter;
+                }
+                text: people.populated && person ? person.secondaryName : ""
+                color: highlighted ? Theme.secondaryHighlightColor : secondaryColor
+                truncationMode: TruncationMode.Fade
+                width: Math.min(implicitWidth, parent.width - firstNameText.width)
+                visible: width > 0
+            }
+
+            Label {
+                text: people.populated && person ? main.formatNumber(remoteUid): ""
+                anchors {
+                    left: parent.left;
+                    top: firstNameText.bottom;
+                    verticalCenter: parent.verticalCenter;
+                }
+                color: highlighted ? Theme.highlightColor : primaryColor
+                font.pixelSize: Theme.fontSizeExtraSmall
+                truncationMode: TruncationMode.Fade
+                width: Math.min(implicitWidth, parent.width)
+                visible: width > 0
+            }
 
-            color: highlighted ? Theme.highlightColor : primaryColor
-            truncationMode: TruncationMode.Fade
-            width: Math.min(implicitWidth, parent.width)
-        }
-        Label {
-            text: people.populated && person ? person.secondaryName : ""
-            color: highlighted ? Theme.secondaryHighlightColor : secondaryColor
-            truncationMode: TruncationMode.Fade
-            width: Math.min(implicitWidth, parent.width - firstNameText.width)
-            visible: width > 0
         }
+
     }
 
     Column {
